tests/components/recorder/conftest.py

```diff
@@ -46,3 +50,70 @@ def recorder_dialect_name(hass: HomeAssistant, db_engine: str) -> Generator[None
             "homeassistant.components.recorder.Recorder.dialect_name", db_engine
         ):
             yield
+
+
+@dataclass
+class InstrumentedMigration:
```

Comment:
```suggestion
class InstrumentedMigration(slots=True):
```

Reply:
This is only used, rarely, in tests :)
But I guess it can't hurt 👍